make/common/NativeCompilation.gmk

```diff
@@ -1145,11 +1186,29 @@ define SetupNativeCompilationBody
 
     $1_TARGET_DEPS := $$($1_ALL_OBJS) $$($1_RES) $$($1_VARDEPS_FILE) $$(STATIC_MAPFILE_DEP)
 
+    $1_AR_OBJ_ARG := $$($1_LD_OBJ_ARG)
+    ifneq ($(findstring $(TOOLCHAIN_TYPE), gcc clang), )
+      $1_TARGET_RELOCATABLE := $$($1_OBJECT_DIR)/$$($1_PREFIX)$$($1_NAME)_relocatable$(OBJ_SUFFIX)
+      $1_AR_OBJ_ARG := $$($1_TARGET_RELOCATABLE)
+    endif
+
     $$($1_TARGET): $$($1_TARGET_DEPS)
+        ifneq ($$($1_OBJ_FILE_LIST), )
+          ifeq ($$($1_LINK_OBJS_RELATIVE), true)
+	    $$(eval $$(call ListPathsSafely, $1_ALL_OBJS_RELATIVE, $$($1_OBJ_FILE_LIST)))
+          else
+	    $$(eval $$(call ListPathsSafely, $1_ALL_OBJS, $$($1_OBJ_FILE_LIST)))
+          endif
+        endif
 	$$(call LogInfo, Building static library $$($1_BASENAME))
 	$$(call MakeDir, $$($1_OUTPUT_DIR) $$($1_SYMBOLS_DIR))
+        ifneq ($(findstring $(TOOLCHAIN_TYPE), gcc clang), )
+	  $$(call ExecuteWithLog, $$($1_OBJECT_DIR)/$$($1_SAFE_NAME)_partial_link, \
+	     $$($1_LD) $(LDFLAGS_CXX_PARTIAL_LINKING) $(LD_OUT_OPTION)$$($1_TARGET_RELOCATABLE) \
+                 $$($1_LD_OBJ_ARG))
```

Comment:
This makes my build pass.
```suggestion
	     $$($1_LD) $(LDFLAGS_CXX_PARTIAL_LINKING) $$($1_SYSROOT_LDFLAGS) \
	         $(LD_OUT_OPTION)$$($1_TARGET_RELOCATABLE) \
                 $$($1_LD_OBJ_ARG))
```

Reply:
Thanks! Tested with a macosx-x86_64 build and also with our prototype on JDK 11 for linux-x64. They still build okay with the change. Looks like it affects/needed for cross build. Committing your fix.